lib/private/Server.php

```diff
@@ -730,7 +730,13 @@ public function __construct($webRoot, \OC\Config $config) {
 
 			if ($config->getSystemValueBool('installed', false) && !(defined('PHPUNIT_RUN') && PHPUNIT_RUN)) {
 				if (!$config->getSystemValueBool('log_query')) {
-					$v = \OC_App::getAppVersions();
+					try {
+						$v = \OC_App::getAppVersions();
+					} catch (\Doctrine\DBAL\Exception $e) {
+						// Database service probably unavailable
+						// Probably related to https://github.com/nextcloud/server/issues/37424
+						$v = [];
```

Comment:
$v = [] means we are using the nextcloud version number as cache prefix. I'm uncertain if we need a cache prefix/a memcache instance after all.  Likely the database is not available, what do we want to cache? 

```suggestion
						return $arrayCacheFactory;
```

I don't know if that's better or worse. Technically, we disable permanent caching when using the arrayCacheFactory and use a temporary cache,

Reply:
I would let you decide on that since you have more experience with this code base and I would implement the conclusion.

Thanks @kesselb

Reply:
Fair question. I think we should return the arrayCacheFactory when the database is unavailable (and we run into the exception block).

What would we cache when the database is unavailable? ;) If the database is available again, the cache prefix is different, hence the old data is stale. A fluent cache provide seems reasonable for this scenario.

Reply:
Thanks @kesselb would do this right away to this can be closed ASAP!

I hear you are not well, I wish you fast recovery! :)